src/jit/morph.cpp

```diff
@@ -7237,17 +7237,517 @@ bool Compiler::fgCanFastTailCall(GenTreeCall* callee)
 #endif
 }
 
+//------------------------------------------------------------------------
+// fgMorphPotentialTailCall: Attempt to morph a call that the importer has
+// identified as a potential tailcall to an actual tailcall.
+//
+// Arguments:
+//    call - The call to morph.
+//    newNode - The new node to use if this function returns true.
```

Comment:
Why do you prefer to return `bool` and pass `GenTree** newNode` instead of returning `GenTree*` that is set to nullptr if morph failed?

If you want to keep this variant we probably need an assert that `newNode != nullptr`.

Reply:
Morph does not really fail for this function -- for a candidate that we cannot tailcall we still "morph" it by removing the "is candidate" flags.

I have no strong preference one way or the other -- I just started out with `bool fgMorphPotentialTailCall(GenTreeCall* call)` to begin with and at some point realized I needed to output the placeholder node. I can change it.

Reply:
The proposed change sounds good to me.